src/reflect/scala/reflect/internal/util/Position.scala

```diff
@@ -133,14 +133,17 @@ private[util] trait InternalPositionImpl {
   final def makeTransparentIf(cond: Boolean): Position =
     if (cond && isOpaqueRange) Position.transparent(source, start, point, end) else this
 
-  /** Copy a range position with a changed value.
-   */
+  /* Copy a range position with a changed value. */
+  /* Note: the result is validated (start <= end), use `copyRange` to update both at the same time. */
   def withStart(start: Int): Position          = copyRange(start = start)
   def withPoint(point: Int): Position          = if (isRange) copyRange(point = point) else Position.offset(source, point)
   def withEnd(end: Int): Position              = copyRange(end = end)
   def withSource(source: SourceFile): Position = copyRange(source = source)
   def withShift(shift: Int): Position          = Position.range(source, start + shift, point + shift, end + shift)
 
+  def copyRange(start: Int = start, point: Int = point, end: Int = end, source: SourceFile = source) =
+    Position.range(source, start, point, end)
```

Comment:
Starting with `source` would effectively encourage use of named args.

Reply:
Although I love named args, I think this is fine, especially for a bug fix. I'd like a lint for APIs with multiple args of the same type. It's possible to use deprecatedName to discourage using a param name, but not conversely to require it.